packages/web/modals/variants-conversion.tsx

```diff
@@ -0,0 +1,363 @@
+  variant: AssetVariant;
+  showBottomBorder?: boolean;
+  onDoneConverting: () => void;
+}> = observer(({ variant, showBottomBorder = true, onDoneConverting }) => {
+  const { t } = useTranslation();
+  const { logEvent } = useAmplitudeAnalytics();
+  const { accountStore } = useStore();
+  const account = accountStore.getWallet(accountStore.osmosisChainId);
+
+  const {
+    onConvert,
+    quote,
+    convertFee,
+    isLoading,
+    isError,
+    isConvertingVariant,
+    isConvertingThisVariant,
+  } = useConvertVariant(variant);
+
+  /**
+   * Check for error when estimating, in case this variant
+   * has very low liquidity and would fail on chain for some reason.
+   * If so, block the user from sending tx.
+   */
+  const simulation = useAsync(async () => {
+    if (!account || !quote || !account?.address) return;
+
+    const messages = await getConvertVariantMessages(
+      variant,
+      quote,
+      account.address
+    );
+
+    if (!Array.isArray(messages)) return;
+
+    await accountStore.estimateFee({ wallet: account, messages }).catch((e) => {
+      logEvent([
+        EventName.ConvertVariants.variantUnavailable,
+        { fromToken: variant.amount.currency.coinDenom },
+      ]);
+
+      throw e;
+    });
+  });
+
+  const isUnavailable = Boolean(simulation?.error) || isError;
+
+  const conversionDisabled =
+    isLoading ||
+    isError ||
+    isConvertingVariant ||
+    simulation.loading ||
+    Boolean(simulation?.error);
+
+  const isButtonLoading = isConvertingThisVariant || simulation.loading;
+
+  // Extracted to add spans to style/differentiate fee details
+  const FeeContent = () => {
+    if (isLoading || !convertFee || !quote?.swapFee || simulation.loading)
+      return <SkeletonLoader className="h-4 w-20" />;
+
+    if (isUnavailable)
+      return (
+        <span className="caption text-rust-300">
+          {t("assetVariantsConversion.unavailable")}
+        </span>
+      );
+
+    // Free
+    if (quote.swapFee.toDec().isZero()) {
+      return (
+        <>
+          {`${t("assetVariantsConversion.conversionFees")}: `}
+          <span className="text-bullish-400">
+            {t("assetVariantsConversion.free")}
+          </span>
+        </>
+      );
+    }
+
+    /**
+     * Put an upper bound on the convert fee since
+     * low liq variants can have wildly high prices if the pool is imbalanced.
+     */
+    const showConvertFee =
+      convertFee.toDec().isPositive() && convertFee.toDec().lt(new Dec(10_000));
+
+    // Fees
+    return (
+      <>
+        {`${t("assetVariantsConversion.conversionFees")}: `}
+        {showConvertFee && (
+          <>
+            <span className="text-white-100">{`${convertFee}`}</span>{" "}
+          </>
+        )}
+        <span className="text-osmoverse-500">
+          {showConvertFee
+            ? `(${quote.swapFee.maxDecimals(2)})`
+            : quote.swapFee.maxDecimals(2).toString()}
+        </span>
+      </>
+    );
+  };
```

Comment:
We should probably move this to a separate component to avoid unnecessary re-renders.